azure/services/networkinterfaces/spec_test.go

```diff
@@ -311,12 +321,14 @@ func TestParameters(t *testing.T) {
 				g.Expect(result.(network.Interface)).To(Equal(network.Interface{
 					Location: to.StringPtr("fake-location"),
 					InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
+						Primary:                     nil,
 						EnableAcceleratedNetworking: to.BoolPtr(true),
 						EnableIPForwarding:          to.BoolPtr(true),
 						IPConfigurations: &[]network.InterfaceIPConfiguration{
 							{
 								Name: to.StringPtr("pipConfig"),
 								InterfaceIPConfigurationPropertiesFormat: &network.InterfaceIPConfigurationPropertiesFormat{
+									Primary:                         to.BoolPtr(true),
```

Comment:
Could we add some tests where we check the new ipConfigurations generated in `spec.go`?